connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/ConnectorUtil.java

```diff
@@ -44,24 +45,23 @@ public final class ConnectorUtil {
   private ConnectorUtil() {}
 
   public static Optional<OutboundConnectorConfiguration> getOutboundConnectorConfiguration(
-      Class<?> cls) {
+      Class<? extends OutboundConnectorFunction> cls) {
     return Optional.ofNullable(cls.getAnnotation(OutboundConnector.class))
         .map(
             annotation -> {
               final var configurationOverrides =
                   new ConnectorConfigurationOverrides(annotation.name(), System::getenv);
-
               return new OutboundConnectorConfiguration(
                   annotation.name(),
                   getInputVariables(cls, annotation),
                   configurationOverrides.typeOverride().orElse(annotation.type()),
-                  cls,
+                  () -> ConnectorHelper.instantiateConnector(cls),
                   configurationOverrides.timeoutOverride().orElse(null));
             });
   }
 
   public static OutboundConnectorConfiguration getRequiredOutboundConnectorConfiguration(
-      Class<?> cls) {
+      Class<? extends OutboundConnectorFunction> cls) {
```

Comment:
Do we need two functions? Can we merge `getOutboundConnectorConfiguration` with `getRequiredOutboundConnectorConfiguration`?